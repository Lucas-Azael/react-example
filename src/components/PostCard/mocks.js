export const postCardPropsMock = {
    title: 'Title mock',
    body: 'Body mock',
    id: 1,
    cover: 'img/mock.png'
}