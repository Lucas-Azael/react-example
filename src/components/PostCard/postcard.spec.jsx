import { render, screen } from "@testing-library/react"
import { PostCard } from '.'
import { postCardPropsMock } from "./mocks"

const props = postCardPropsMock;

describe('<PostCard />', () => {
    it('should render component', () => {
       render(<PostCard {...props} />)

       expect(screen.getByRole('img', { name: 'Title mock'}))
       .toHaveAttribute('src', props.cover);

       expect(screen.getByRole('heading', {name: 'Title mock'})).toBeInTheDocument()

       expect(screen.getByText('Body mock')).toBeInTheDocument();
    });

    it('should match snapshot', () => {
        const { container } = render(<PostCard {...props} />);
        //O primeiro filho do container é o elemento que ele está renderizando
        expect(container.firstChild).toMatchSnapshot();
    });
})