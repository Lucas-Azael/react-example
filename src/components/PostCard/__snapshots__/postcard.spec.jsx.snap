// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="Title mock"
    src="img/mock.png"
  />
  <div
    class="post-content"
  >
    <h1>
      Title mock
    </h1>
    <p>
      Body mock
    </p>
  </div>
</div>
`;
